### PR TITLE
Softmax client example

### PR DESCRIPTION
**Pull request by aosewski**

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/device_softmax.hpp [hunk 1]
```diff
@@ -24,227 +15,54 @@ namespace device {
...
-          index_t KThreadClusterSize,
-          index_t MThreadSliceSize,
-          index_t KThreadSliceSize,
-          index_t InSrcVectorDim,
-          index_t InSrcVectorSize,
-          index_t OutDstVectorSize>
-struct DeviceSoftmax : public DeviceNormalization
+          typename InElementwiseOp,
+          typename AccElementwiseOp,
+          index_t Rank>
```
Comment (asroy) [inline include/ck/tensor_operation/gpu/device/device_softmax.hpp:20]: Please use ```NDim``` or ```NumDim``` to be consistent

**Review by asroy — APPROVED** [review]

**Review by asroy — DISMISSED** [review]

**Comment by aosewski** [comment]
Waiting for CI to be up.

**Comment by rosenrodt** [comment]
I see two different file organization styles:

Layernorm:
device op interface in `device_normalization.hpp` 
implementation in `device_layernorm_impl.hpp`. 

Softmax after this PR:
device op interface in `device_softmax.hpp`
implementation in `impl/device_softmax_impl.hpp` (extra nested directory)

Can we harmonize these two file organization styles?

**Comment by aosewski** [comment]
@rosenrodt Currently both implementations and device op interface are in single directory which IMHO makes a mess. I though that it would make cleaner file organization if we could move all implementations inside ie. `impl` directory. Thus this PR is a proposition of such changes. 
As for the naming of interface file and impl for layernorm and softmax I agree it's awkward right now. Probably because of the fact that both files lies within single directory, thus it's hard to name them meaningfully. Actually I'd name all device op's interface files as the device op and follow our current naming of kernel files which add some suffix to the device op name indicating implementation details.
I'd propose to make those refactoring and renaming in separate PR. What do you think about it?  @asroy

**Review by asroy — APPROVED** [review]

**Pull request merged**